src/gpu/amd/sycl_hip_utils.hpp

```diff
@@ -131,14 +131,19 @@ inline status_t convert_data_type(const memory_desc_t *mem_desc,
         case data_type_t::dnnl_bf16:
             *miopen_data_type = miopenDataType_t::miopenBFloat16;
             break;
-        case data_type_t::dnnl_s8:
-            *miopen_data_type
-                    = ((vectorized
-                               && mem_desc->format_desc.blocking.inner_blks[0]
-                                       == 4)
-                                    ? miopenDataType_t::miopenInt8x4
-                                    : miopenDataType_t::miopenInt8);
+        case data_type_t::dnnl_s8: {
+            if (vectorized
+                    && mem_desc->format_desc.blocking.inner_blks[0] == 4) {
+#if MIOPEN_HAS_INT8X4
+                *miopen_data_type = miopenDataType_t::miopenInt8x4;
+#else
+                return status::unimplemented;
+#endif
+            } else {
+                *miopen_data_type = miopenDataType_t::miopenInt8;
+            }
```

Comment:
Would `miopenDataType_t::miopenInt8` work for `vectorized && mem_desc->format_desc.blocking.inner_blks[0] == 4` condition?
If yes, maybe change the scope of the macro to avoid `unimplemented` when the condition is hit...

Reply:
I had a look at convolution (as the miopen documentation mentions int8x4 is used primarily for convolution forward pass) and with the current implementation I believe it won't work as an miopen blocked format would need to be used for those cases, which is currently not supported in the oneDNN miopen implementations. I think it's okay to return unimplemented for now and when blocked format support is added this could be enabled, but let me know what you think @dzarukin .

Reply:
I'm fine with the solution, just tried to iterate if we could save people (if there are any) from upgrading the ROCm version and getting surprise unimplemented statuses in cases which worked before.